### PR TITLE
Add big support table

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Add big support table

##### ISSUE TYPE
- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by samccann** [comment]
Great information here! Sadly the table is too wide:
image

A couple of options:
 - change `General Availability/General Support` to just `General Availability` ? That would shrink down one column a bit.
 - Combine the python.powershell details into one column.

Another alternative - flip the table on its 'side' with the version numbers as column headings and the current column headings as rows.  We'd want to only include the most-recently EOL'd release or again, the table gets too wide.

We also need to consider how often we repeat the python details. It's already here - https://docs.ansible.com/ansible-core/devel/installation_guide/intro_installation.html#managed-node-requirements. So I'm not sure we want to repeat it in this section. We could just put a cross-ref to that table and add the powershell details...

**Comment by sivel** [comment]
> Sadly the table is too wide

Can you offer guidance about what width is better?

> * change `General Availability/General Support` to just `General Availability` ? That would shrink down one column a bit.

I can also stack the values so it's not wide, I'm doing that in other cells purposefully

> We also need to consider how often we repeat the python details

Yeah, I didn't really want that table when it was added, but it seemed like people were too anxious to get it into docs.  I'd rather document it all in one convenient table, and point to that table.  So my preference would be to remove the other table.  That was my intention from the beginning, but the other table got added before I could finish this one.

**Comment by sivel** [comment]
Ok, I removed `General Support`, and combined the `Target` columns

**Comment by samccann** [comment]
What do you think about combining critical/security/eol into one column called security, and then each cell would be something like:
 critical - May 2023
security - Nov 2023
EOL - May 2024


That might get it down to a  size of five columns.  Judging when a table is 'too wide' is basically guesswork. In general,  these tables will revert to horizontal scrolling  on anything that isn't a computer screen (aka tablets, phonesl).  Shrink down your browser at http://docs.testing.ansible.com/ansible-core/2.15/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix to get an idea of what I'm talking about.  

... and pls don't shoot the messenger - but I just looked at that table on my phone and I think it would be better to flip the order of the rows (putting most recent version 2.15 at the top and then descending order on the versions). That way, the most recent and still relevent sections are still visible with the column headers even on my phone.
image

**Comment by webknjaz** [comment]
@sivel radical suggestion — has anybody ever considered having an in-project sphinx extension for generating such a table from a structured YAML file? It seems like that could be easier to maintain. I have experience in implementing such things if a consult is needed, it's quite simple.

**Comment by samccann** [comment]
@sivel - since I'm getting picky on the table output, I'm happy to take this over next week once ansible-core releases and I'm done with those docs...

**Comment by samccann** [comment]
@sivel - I experimented locally to shrink the number of columns. Let me know what you think:

image (4)

File: docs/docsite/rst/reference_appendices/release_and_maintenance.rst [hunk 1]
```diff
@@ -115,52 +115,181 @@ You can refer to the :ref:`core_porting_guides` for tips on updating your playbo
...
+Dates listed indicate the start date of the maintenance cycle.
+
+.. list-table::
+   :header-rows: 1
+    
+   * - Version
+     - Support
+     - End Of Life
+     - Controller Python
+     - Target Python / Powershell
```
Comment (mattclay) [inline docs/docsite/rst/reference_appendices/release_and_maintenance.rst]:
```suggestion
     - Target Python / PowerShell
```

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Can't believe I held this PR up for 2 months!  anyway, thanks for your patience and merging this now.

**Pull request merged**